x-pack/platform/plugins/shared/fleet/public/applications/integrations/sections/epm/screens/detail/policies/package_policies.tsx

```diff
@@ -122,11 +133,27 @@ export const PackagePoliciesPage = ({
     }`,
   });
   useEffect(() => {
-    setAgentBasedPackageAndAgentPolicies(
-      !agentBasedData?.items ? [] : agentBasedData.items.map(mapPoliciesData)
-    );
+    const mappedPoliciesData = !agentBasedData?.items
+      ? []
+      : agentBasedData.items.map(mapPoliciesData);
+    setAgentBasedPackageAndAgentPolicies(mappedPoliciesData);
   }, [agentBasedData, mapPoliciesData]);
 
+  const isInputPackageDatasetUsedByMultiplePolicies = useCallback(
```

Comment:
I think this will probably not work with space awareness enabled, as the dataset could be used in policies in a different space, I think here one solution could be to put the responsability on the user and ask him if he want to delete the related assets and explain if other policies use that template it will be problematic wdyt?

Reply:
Probably having that check in the API is better, and we can handle this nicely in the UI, with a warning `assets for that package policy has not been deleted as they are used by other package policy ...`

Reply:
> Probably having that check in the API is better, and we can handle this nicely in the UI, with a warning assets for that package policy has not been deleted as they are used by other package policy ...

it looks like a sensible way to handle it. I'll check this solution and see if there are any caveats.